cocos/editor-support/cocostudio/ActionTimeline/CCFrame.cpp

```diff
@@ -618,20 +618,44 @@ void InnerActionFrame::onEnter(Frame *nextFrame, int currentFrameIndex)
 
 void InnerActionFrame::setStartFrameIndex(int frameIndex)
 {
+#ifdef CC_STUDIO_ENABLED_VIEW   // for cocostudio only
+    if (_enterWithName)
+    {
+        CCLOG(" cannot set start when enter frame with name. setEnterWithName false firstly!");
+        throw std::exception();
+    }
+#else
     CCASSERT(!_enterWithName, " cannot setStartFrameIndex when enterWithName is set");
+#endif
     _startFrameIndex = frameIndex;
 }
 
 
 void InnerActionFrame::setEndFrameIndex(int frameIndex)
 {
+#ifdef CC_STUDIO_ENABLED_VIEW   // for cocostudio only
+    if (_enterWithName)
+    {
+        CCLOG(" cannot set end when enter frame with name. setEnterWithName false firstly!");
+        throw std::exception();
+    }
+#else
     CCASSERT(!_enterWithName, " cannot setEndFrameIndex when enterWithName is set");
+#endif
     _endFrameIndex = frameIndex;
 }
 
 void InnerActionFrame::setAnimationName(const std::string& animationName)
 {
+#ifdef CC_STUDIO_ENABLED_VIEW   // for cocostudio only
+    if (!_enterWithName)
+    {
+        CCLOG(" cannot set aniamtioname when enter frame with index. setEnterWithName true firstly!");
+        throw std::exception();
```

Comment:
Please follow the coding style: https://github.com/cocos2d/cocos2d-x/wiki/Cpp--Coding-Style#exceptions